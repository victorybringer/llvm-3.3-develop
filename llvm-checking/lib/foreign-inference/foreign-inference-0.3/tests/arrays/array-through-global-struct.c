/*
  I've changed my mind on this test.  Doing a shape analysis to
  note that s.fld aliases arr1 is too expensive.  Do not catch this
  at this point in the analysis.

  This can be changed later and be found by noting that field S.fld is
  used as an array in various points in the application and the
  assignment of arr1 to s.fld makes arr1 an array.
 */
struct S {
  int * fld;
  double d;
};

extern int x;
struct S s;

int f(int *arr1, double ** arr2) {
  s.fld = arr1;
  s.d = arr2[6][x];
  return s.fld[x] / s.d;
}
