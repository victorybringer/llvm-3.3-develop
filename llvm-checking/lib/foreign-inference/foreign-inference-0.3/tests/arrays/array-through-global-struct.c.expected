fromList [ (("@f", "%arr2"), 2)
         , (("@f", "%arr1"), 1)
         ]
